### PR TITLE
fix: lockfile corrupt after config dep removal sometimes

### PR DESCRIPTION
**Pull request by dsherret**
For https://github.com/denoland/deno/issues/22250

### CODE REVIEW

File: tests/specs/config_changes/DenoIssue22250.txt
Comment (dsherret) [inline tests/specs/config_changes/DenoIssue22250.txt:1]: See the diff in https://github.com/denoland/deno_lockfile/commit/e3870c581f0c1c6d952ed0e1f34aa9569bbec5bc for how this file changed

File: tests/integration_test.rs [hunk 1]
```diff
@@ -130,13 +131,38 @@ fn config_changes() {
...
 }
 
+fn verify_packages_content(packages: &PackagesContent) {
+  // verify the specifiers
+  for id in packages.specifiers.values() {
+    if let Some(npm_id) = id.strip_prefix("npm:") {
+      assert!(packages.npm.contains_key(npm_id), "Missing: {}", id);
+    } else if id.strip_prefix("jsr:").is_some() {
+      // ignore jsr packages because they won't be in the lockfile when they don't have dependencies
+      // todo(dsherret): actually include them here because we need to lock the manifest version
```
Comment (bartlomieju) [inline tests/integration_test.rs:149]: Open an issue about it?

**Review by bartlomieju — APPROVED** [review]
LGTM

File: tests/integration_test.rs (hunk 1 above)
Comment (bartlomieju) [reply tests/integration_test.rs:149]: Nvm, I see it

**Pull request merged**